core/server/common/src/main/java/alluxio/master/metastore/Metastore.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metastore;
+
+/**
+ * A store for managing Alluxio metadata.
+ */
+public interface Metastore {
+  /**
+   * @return the file store
+   */
+  FileStore getFileStore();
+
+  /**
+   * @return the block store
+   */
+  BlockStore getBlockStore();
+
+  /**
+   * Metastore types.
+   */
+  enum Type {
+    HEAP,
+    ROCKS
```

Comment:
If I want to create a new `Master` which needs metadata, I need to implement both the heap and rocks version of that metadata?

Reply:
The way this is handled for `MetaMaster` and `MetricsMaster` is that they use their own in-memory data structures instead of depending on the `Metastore`